third_party/kiwi/kiwi/var.h

```diff
@@ -46,6 +47,18 @@ kiwi_var_init(kiwi_var_t *var, char *name, int name_len)
 	var->value_len = 0;
 }
 
+static inline void
+kiwi_var_init_with_type(kiwi_var_t *var,
+                        char *name,
+                        int name_len,
+                        kiwi_var_type_t type)
+{
+	var->type      = type;
```

Comment:
Is this type necessary?

Reply:
Yes, without the correct type `kiwi_vars_cas` don't work correctly. When `kiwi_vars_cas` called `kiwi_var_compare` type mismatch happens.